weasyl/middleware.py

```diff
@@ -19,6 +23,12 @@
 from weasyl.error import WeasylError
 
 
+def _get_headers(env):
+    return {
+        key[5:].replace('_', '-').title(): value
+        for key, value in env.iteritems() if key.startswith('HTTP_')}
```

Comment:
Yep, [it's in `weasyl.oauth2`](https://github.com/Weasyl/weasyl/blob/4ad53b59da2fd7452546f04025414780911190f2/weasyl/oauth2.py) as `extract_params`.